react/test/index.ts

```diff
@@ -674,3 +674,11 @@ class SyntheticEventTargetValue extends React.Component<{}, { value: string }> {
     });
   }
 }
+
+type DivAttrs = React.DOMAttributes<HTMLDivElement>;
+type DivFactory = React.DOMFactory<DivAttrs, HTMLDivElement>;
+
+type DivProps = React.HTMLProps<HTMLDivElement>;
+type DivPropsBase = React.ChangeTargetHTMLProps<HTMLDivElement>;
+const d1:DivProps = undefined as any;
+const d2:DivPropsBase = d1;
```

Comment:
These tests don't seem to test anything particularly useful. I think something like

```ts
React.DOM.input({
    onChange: event => {
        event.target.value; // `event.target` is guaranteed to be HTMLInputElement
    }
});
```

would be more appropriate.